src/Microsoft.AspNet.Routing/VirtualPathData.cs

```diff
@@ -0,0 +1,66 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNet.Routing
+{
+    /// <summary>
+    /// Represents information about the route and virtual path that are the result of
+    /// generating a URL with the ASP.NET routing middleware.
+    /// </summary>
+    public class VirtualPathData
+    {
+        private string _virtualPath;
+        private readonly RouteValueDictionary _dataToken;
```

Comment:
Since its actual type is RouteValueDctionary I used that one. I think it should be IReadOnlyDctionary<string, object> here since there is no need to change the dictionary from user codes.